File: gdx2d-demoDesktop/src/hevs/gdx2d/demos/physics/pinball/SensorSpinner.java
package hevs.gdx2d.demos.physics.pinball;

import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.interfaces.DrawableObject;
import hevs.gdx2d.lib.physics.AbstractPhysicsObject;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.math.Vector2;

public class SensorSpinner extends Sensor implements DrawableObject {
	
	protected TextureRegion[] sprites;
	protected float speed;
	protected float pos;
	private float width;
	private float height;

	public SensorSpinner(String name, Vector2 position, float width, float height, TextureRegion sprites[]) {
		super(name, position, width, height);
		this.sprites = sprites;
		this.width = width;
		this.height = height;
	}

	@Override
	public void draw(GdxGraphics g) {
		pos += speed;
		speed -= speed * 0.8f*Gdx.graphics.getDeltaTime();
		int i = ((int) (pos)) % sprites.length;
		
		float w = sprites[i].getRegionWidth();
		float h = sprites[i].getRegionHeight();
		g.draw(sprites[i], getBodyPosition().x-w/2, getBodyPosition().y-h/2, w/2, h/2, w, h, width/w, height/h, 0);
	}
	
	@Override
	public void collision(AbstractPhysicsObject theOtherObject, float energy) {
		super.collision(theOtherObject, energy);
		System.out.println(getClass()+" uohoho "+ theOtherObject.getBody().getLinearVelocity());
		speed = theOtherObject.getBody().getLinearVelocity().len()/2;
	}
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/physics/pinball/SensorSpinner.scala
package ch.hevs.gdx2d.demos.physics.pinball

import ch.hevs.gdx2d.lib.GdxGraphics
import ch.hevs.gdx2d.lib.interfaces.DrawableObject
import ch.hevs.gdx2d.lib.physics.AbstractPhysicsObject
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.math.Vector2

class SensorSpinner(name: String, position: Vector2, private val width: Float, private val height: Float, protected var sprites: Array[TextureRegion])
  extends Sensor(name, position, width, height) with DrawableObject {

  protected var speed: Float = 0f
  protected var pos: Float = 0f

  override def draw(g: GdxGraphics): Unit = {
    pos += speed
    speed -= speed * 0.8f * Gdx.graphics.getDeltaTime
    val i = pos.toInt % sprites.length

    val w = sprites(i).getRegionWidth.toFloat
    val h = sprites(i).getRegionHeight.toFloat
    g.draw(sprites(i), getBodyPosition.x - w / 2, getBodyPosition.y - h / 2, w / 2, h / 2, w, h, width / w, height / h, 0f)
  }

  override def collision(theOtherObject: AbstractPhysicsObject, energy: Float): Unit = {
    super.collision(theOtherObject, energy)
    speed = theOtherObject.getBody.getLinearVelocity.len() / 2f
  }
}
